core/src/main/java/org/ehcache/spi/ServiceLocator.java

```diff
@@ -164,25 +164,28 @@ void addService(final Service service, final boolean expectsAbstractRegistration
     return interfaces;
   }
 
-  @Override
-  public <T extends Service> T findServiceFor(ServiceConfiguration<T> config) {
-    return findService(config.getServiceType(), config);
+  public <T extends Service> T getOrCreateServiceFor(ServiceConfiguration<T> config) {
+    return findService(config.getServiceType(), config, true);
   }
 
   @Override
-  public <T extends Service> T findService(Class<T> serviceType) {
-    return findService(serviceType, null);
+  public <T extends Service> T getService(Class<T> serviceType) {
+    return findService(serviceType, null, false);
+  }
+
+  public <T extends Service> T getOrCreateService(Class<T> serviceType) {
```

Comment:
Crazy idea... could we eliminate this method entirely by adding services that list the mandatory services as their dependencies?